azure/services/managedclusters/managedclusters.go

```diff
@@ -184,240 +70,57 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "managedclusters.Service.Reconcile")
 	defer done()
 
-	managedClusterSpec, err := s.Scope.ManagedClusterSpec()
-	if err != nil {
-		return errors.Wrap(err, "failed to get managed cluster spec")
-	}
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
 
-	isCreate := false
-	existingMC, err := s.Client.Get(ctx, managedClusterSpec.ResourceGroupName, managedClusterSpec.Name)
-	// Transient or other failure not due to 404
-	if err != nil && !azure.ResourceNotFound(err) {
-		return azure.WithTransientError(errors.Wrap(err, "failed to fetch existing managed cluster"), 20*time.Second)
+	managedClusterSpec := s.Scope.ManagedClusterSpec(ctx)
+	if managedClusterSpec == nil {
+		return nil
```

Comment:
Maybe add a log statement like in the other async services?

Reply:
ditto